src/targets/gpu/gemm_impl.cpp

```diff
@@ -99,180 +115,477 @@ static bool is_transposed(const shape& s)
+            auto ded_args = pack(rocblas_gemm_algo_solution_index, int8_flag, nullptr, &list_size);
+
+            rocblas_invoke(&rocblas_gemm_strided_batched_ex_get_solutions,
+                           pack_join(common_args, ded_args));
+            solution_indices.resize(list_size);
+
+            auto common_sol_args = create_strided_batched_args_common(ctx, input_args);
+            auto ded_sol_args    = pack(
+                rocblas_gemm_algo_solution_index, int8_flag, solution_indices.data(), &list_size);
+
+            rocblas_invoke(&rocblas_gemm_strided_batched_ex_get_solutions,
+                           pack_join(common_sol_args, ded_sol_args));
+        }
+        else
+        {
+            auto common_args = create_gemm_ex_args_common(ctx, input_args);
+            auto ded_args = pack(rocblas_gemm_algo_solution_index, int8_flag, nullptr, &list_size);
+            rocblas_invoke(&rocblas_gemm_ex_get_solutions, pack_join(common_args, ded_args));
+            solution_indices.resize(list_size);
+
+            auto common_sol_args = create_gemm_ex_args_common(ctx, input_args);
+            auto ded_sol_args    = pack(
+                rocblas_gemm_algo_solution_index, int8_flag, solution_indices.data(), &list_size);
+            rocblas_invoke(&rocblas_gemm_ex_get_solutions,
+                           pack_join(common_sol_args, ded_sol_args));
+        }
+
+        double bestTime   = std::numeric_limits<double>::max();
+        double first_time = -1;
+        // Initialize to default solution index
+        rocblas_int bestSol = 0;
+        for(auto sol : solution_indices)
+        {
+            // Warmup: the first few calls to an op. may not be representative since there is
+            // more time taken initializing caches, etc. so we won't time them.
+            for(rocblas_int cc = 0; cc < cold_calls; ++cc)
+            {
+                run(ctx, input_args, sol);
+            }
+            double host_time = 0.0;
+            // Define the function to be timed
+            auto run_func = [&]() {
+                run(ctx, input_args, sol);
+                ctx.finish();
+            };
+            for(rocblas_int hc = 0; hc < hot_calls; ++hc)
+            {
+                ctx.finish();
+                host_time += time<microseconds>(run_func);
+            }
+            // todo:  Measured time dropped from 20 us to about 6.7 us when I raised hot_calls from
+            // 1 to 11. The higher the hot_calls value, the faster per-call time up to at least 25,
+            // and increasing cold_calls makes little or no difference.  Why?
+            host_time /= hot_calls;
+
+            // debug only: track time for first solution.
+            if(first_time < 0)
+                first_time = host_time;
+
+            // track current best
+            if(host_time < bestTime)
+            {
+                printf(" current best index  %d, time %g\n", sol, host_time);
```

Comment:
Use `std::cout`.

Reply:
done.  Not sure if the message is needed at all.